HW2/P2/parallel_vector.pyx

```diff
@@ -105,11 +145,55 @@ cpdef move_data_medium_grained(np.int32_t[:] counts,
    # to parallelize data movement.  Be sure to avoid deadlock, as well as
    # double-locking.
    ##########
-   with nogil:
-       for r in range(repeat):
-           for idx in range(src.shape[0]):
-               if counts[src[idx]] > 0:
+
+   for r in prange(repeat, nogil=True, num_threads=threads): #comment to make work
+      
+       for idx in range(src.shape[0]):
+
+           if counts[src[idx]] > 0:
```

Comment:
should acquire lock before accessing counts! because counts could be changing in the meantime!